packages/core/src/drawer/drawer.ts

```diff
@@ -0,0 +1,110 @@
+import type { TemplateResult } from 'lit-html';
+import type { CSSResultGroup } from 'lit';
+import type { ButtonSize } from '../button/button.js';
+import { LitElement, html } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import styles from './drawer.scss.js';
+
+export type DrawerAttachement = 'right' | 'left' | 'top' | 'bottom';
+
+/**
+ * A dialog component for displaying modal UI.
+ *
+ * @slot default - Body content for the dialog
+ * @slot header - Header content for the dialog
+ * @slot title - The title of the dialog
+ */
+export class Drawer extends LitElement {
```

Comment:
So since you're using other SLDS web components inside the drawer, you should use `ScopedElementsMixin`; otherwise you may run into conflicts when 2 components try to define the same child web component.